pkg/app/app_wasm.go

```diff
@@ -139,7 +139,18 @@ func navigate(u *url.URL, updateHistory bool) error {
 
 	root, ok := routes[path]
 	if !ok {
-		root = NotFound
+		for i := range routesRe {
+			s := routesRe[i].re.FindString(path)
```

Comment:
Since you are not using the resulting string, you could use the [MatchString()](https://golang.org/pkg/regexp/#Regexp.MatchString) method.

```go
	if !r.regexp.MatchString(path) {
		continue
	}
```

Reply:
I tried that, but it fails one test case. I'm not sure why, exactly. I left one line commented out so you can try it and run the test suite again to see the result. The code that I checked in on the regexp-routes branch has a couple additional test cases, and I added the t.Helper() so it's easier to tell from the logs which test case fails with MatchString().

Reply:
here's more info on the use case...
the regexp url handler has a folder of encrypted web pages. it  uses the url to figure out the document to load and decrypt, and then will insert the Raw() html into the page DOM. Before inserting the page, it rewrites all href links to make sure they conform with the regexp pattern so that it gets called to decrypt the next page...